pkg/block/mem/adapter.go

```diff
@@ -24,62 +25,88 @@ var (
 	ErrNoPropertiesForKey = fmt.Errorf("no properties for key")
 )
 
+type partInfo struct {
+	data         []byte
+	lastModified time.Time
+	size         int64
+}
+
 type mpu struct {
-	id    string
-	parts map[int][]byte
+	id        string
+	objectKey string // the key (from getKey(obj))
+	initiated time.Time
+	parts     map[int]partInfo // part number -> partInfo
 }
 
-func newMPU() *mpu {
+func newMPU(objectKey string) *mpu {
 	uid := uuid.New()
 	uploadID := hex.EncodeToString(uid[:])
 	return &mpu{
-		id:    uploadID,
-		parts: make(map[int][]byte),
+		id:        uploadID,
+		objectKey: objectKey,
+		initiated: time.Now(),
+		parts:     make(map[int]partInfo),
 	}
 }
 
 func (m *mpu) get() []byte {
```

Comment:
Technically this function should lock - a caller may upload another part while completing.  They must either succeed or get an error, but with the current code they can crash the server.

Not critical given that this adapter is only used in fakes, but still...